### DISCUSSION TITLE
Conference room data channel only reports audio updates

### DISCUSSION THREAD

**Discussion by VanIseghemThomas** [ant-media/Ant-Media-Server#6274] [Q&A]
Title says it all. I'm currently implementing conference streaming for the Unity platform and everything works like a charm except one aspect. When opening a data channel I am only able to retrieve UPDATE_AUDIO_LEVEL and AUDIO_TRACK_ASSIGNMENT event messages. Causing me to have problems mapping the video tracks of joined users since the I cant get ahold of the VIDEO_TRACK_ASSIGNMENT_LIST for example, even when explicitly querying it using the signaling. This is the case for all other features such as camera, mic mute, chat events etc.

My first thought was that I did something wrong opening the data channel, but I'm clearly getting the audio events so there is in fact a data channel. Any obvious things that could cause this?

**Comment by zerostack0** [comment]
Hi @VanIseghemThomas 

You can use the TRACK_LIST_UPDATED message to be able to get the updates. VIDEO_TRACK_ASSIGNMENT_LIST triggerers only your video track assignment change but it didn't trigger everytime someone enters in a room. 

For example, if the maximum video track count is 3, there is 5 participants in the room and a new participant is joined, you won't get VIDEO_TRACK_ASSIGNMENT_LIST.

You can stick with TRACK_LIST_UPDATED and broadcast_object callbacks. For more information: https://github.com/ant-media/conference-call-application/blob/main/react/src/pages/AntMedia.js#L1379 and https://github.com/ant-media/conference-call-application/blob/main/react/src/pages/AntMedia.js#L699

**Reply by VanIseghemThomas** [reply]
Currently I'm using the broadcast object callbacks to detect changes in subtracks, but I need the tracklist to map the video streams to their labels. This is only possible using the TRACK_LIST_UPDATED event as far as I see for now, still looking in ways to get around this. Also tried `getVideoTrackAssignmentsCommand` over the websocket to retrieve this, but this never returns data.

But I believe in my case there is an underlying larger issue since chat messages and others are also not sent over.

**Reply by VanIseghemThomas** [reply]
Server should be fine, the conference example works like expected and receives the appropriate data channel messages when expected. So most likely there is an issue in my implementation setting up the data channel. But then again I'm confused why I am getting the audio updates. Are these handled over a separate data channel or something? As far as I see not but I could be wrong.

**Reply by mekya** [reply]
Hi @mustafaboleken,
Any update?

**Reply by VanIseghemThomas** [reply]
Hi @mustafaboleken 

Thanks for checking up!
IIRC one of the things I observed from logs is that the ICE connection state starts at `Connected`, returns all the audio updates over the data channel, but then after a little while (let's say about 15 sec.) it goes into the state `Completed` but does not return any data anymore. Maybe that can be a clue for something?

From [the WebRTC docs for the web](https://developer.mozilla.org/en-US/docs/Web/API/RTCPeerConnection/iceConnectionState) which I assume Unity ported it's implementation from, it states: 

> [connected](https://developer.mozilla.org/en-US/docs/Web/API/RTCPeerConnection/iceConnectionState#connected)
A usable pairing of local and remote candidates has been found for all components of the connection, and the connection has been established. It is possible that gathering is still underway, and it is also possible that the ICE agent is still checking candidates against one another looking for a better connection to use.

> [completed](https://developer.mozilla.org/en-US/docs/Web/API/RTCPeerConnection/iceConnectionState#completed)
The ICE agent has finished gathering candidates, has checked all pairs against one another, and has found a connection for all components.

A theory I have is that it switches to another connection causing the data channel to close, but I'm not really an expert at WebRTC.

**Reply by burak-58** [reply]
We may make a call to debug it together.

**Reply by VanIseghemThomas** [reply]
@mustafaboleken @burak-58 we are currently really pushing towards a release and currently our trail license expired. We are looking to get a full license though. The next few weeks will be hard to continue on this, the release is planned for the start of June. After those I will get back to you guys and we can plan a moment to sit together and debug this. Already thanks in advance!

**Reply by VanIseghemThomas** [reply]
@mustafaboleken @burak-58  
We finished our release. I sent Mustafa a mail regarding this.

**Reply by VanIseghemThomas** [reply]
Sent it. Did you receive it @mustafaboleken?

**Reply by VanIseghemThomas** [reply]
@mustafaboleken Just checking up, any updates since our meeting?
